prow/config/jobs.go

```diff
@@ -434,6 +436,50 @@ type UtilityConfig struct {
 	DecorationConfig *prowapi.DecorationConfig `json:"decoration_config,omitempty"`
 }
 
+func (u *UtilityConfig) Validate() error {
+	if u.DecorationConfig != nil && len(u.DecorationConfig.SSHKeySecrets) > 0 {
+		if len(u.CloneURI) == 0 {
+			return errors.New("SSH key secrets provided but no clone_uri has been found")
+		}
+	}
+
+	cloneURIValidate := func(cloneURI string) error {
+		if u.DecorationConfig != nil && len(u.DecorationConfig.SSHKeySecrets) > 0 {
+			// Validate if the URI is in a format that can be used via ssh
+			// Example: git@github.com:kubernetes/test-infra.git
```

Comment:
We should either detect a non http(s) URL and not try to do any validation, or search the [RFC](https://tools.ietf.org/html/rfc4251) for the authoritative validation.

Reply:
Since this is over SSH, for instance, if I have `git` as the default user for `github.com` host, I don't need to provide it and it will work even though your regex will deem it invalid